### PR TITLE
acceptance: migrate to docker/engine-api

### PR DESCRIPTION
**Pull request by tamird**
Commit history is a mess, I'll fix it up before merging.

Chaos test is currently failing on my machine, and I have no idea why.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4282)

### CODE REVIEW

**Comment by tamird** [comment]
Argh, this is a good one. One of the dependencies uses build tags that distinguish linux from OS X, so `glock save ...` does the wrong thing on OS X. I had to resort to `build/builder.sh glock save ...` to get the correct GLOCKFILE generated.

File: acceptance/cluster/docker.go [hunk 1]
```diff
@@ -170,32 +119,18 @@ func (c *Container) Kill() error {
...
 // Pause pauses a running container.
 func (c *Container) Pause() error {
-	return c.cluster.client.PauseContainer(c.ID)
+	return c.cluster.client.ContainerPause(c.id)
 }
 
 // Unpause resumes a paused container.
 func (c *Container) Unpause() error {
-	return c.cluster.client.UnpauseContainer(c.ID)
+	return c.cluster.client.ContainerUnpause(c.id)
```
Comment (tschottdorf) [inline acceptance/cluster/docker.go]: funny how everything just reversed.

**Comment by BramGruneir** [comment]
I gave the code a quick scan, overall looks good, but I'll do an in depth pass tomorrow morning.  What version of docker are you running locally?

**Comment by tamird** [comment]
1.9.1

**Comment by BramGruneir** [comment]
Cool.  I'll test it against my setup with v1.10

**Comment by tschottdorf** [comment]
LGTM, I'm going to try running it now.

---

Reviewed 2 of 4 files at r1, 5 of 5 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/chaos_test.go, line 224 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282#-KABKh41s0Q6XTyXuvSg:-KABKh41s0Q6XTyXuvSh:-858285055) ([raw file](https://github.com/cockroachdb/cockroach/blob/9369735a151854f57a480e5e7c6e42d019c1fc3a/acceptance/chaos_test.go#L224)):*
Feel strongly about this label? They always strike me as nasty. Maybe

```
madeProgress := func() bool {
  c.Assert(t)
  for i, client := .... { ... }
  return false
}

for !madeProgress() {
  time.Sleep(time.Second)
}
```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282#-:-KABMVitD9LcAC5DvMem:-1235539418)*

**Comment by tamird** [comment]
Review status: 4 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/chaos_test.go, line 224 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282#-KABKh41s0Q6XTyXuvSg:-KABUM1h-YLfS6UiVCfp:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/9369735a151854f57a480e5e7c6e42d019c1fc3a/acceptance/chaos_test.go#L224)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282)*

**Comment by tamird** [comment]
OK this is finally green. @tschottdorf or @petermattis can I get sign-off?

**Comment by BramGruneir** [comment]
LGTM

**Comment by BramGruneir** [comment]
Also, passes locally with docker 1.10 for me.

**Comment by tschottdorf** [comment]
Still trying to fix my Docker....... but LGTM

---

Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[acceptance/cluster/docker.go, line 102 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282#-KABoh67lJJUyOzm1Q6o:-KABoh67lJJUyOzm1Q6p:-101521041) ([raw file](https://github.com/cockroachdb/cockroach/blob/69f2d22ff69cb72e4c79a821afed8d400a2c6853/acceptance/cluster/docker.go#L102)):*
I remember that error, and thought ignoring it fixed the issue (it still deleted it (?)). Have you checked?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282#-:-KABosr4zcChHnrwvzhr:-1561420379)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[acceptance/cluster/docker.go, line 102 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282#-KABoh67lJJUyOzm1Q6o:-KABp58q_CQHqJbvFA4H:-274131314) ([raw file](https://github.com/cockroachdb/cockroach/blob/69f2d22ff69cb72e4c79a821afed8d400a2c6853/acceptance/cluster/docker.go#L102)):*
I don't know, actually. Let me try it in a follow-up PR in which I ignore the error.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4282)*

**Pull request merged**